lib/shrinkwrap.js

```diff
@@ -74,6 +74,18 @@ function shrinkwrapDeps (dev, problems, deps, tree, seen) {
         (topname ? ', required by ' + topname : ''))
     }
   })
+  if (dev) {
+    Object.keys(tree.missingDevDeps).forEach(function (name) {
```

Comment:
Can this be extracted into a function that can be used for both `dependencies` and `devDependencies`, similar to what you did for #12846? AFAICT, the only difference is that one says "`invalid:`", and the other says, "`invalid dev dep:`".